core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -146,10 +190,152 @@ private Http2ConnectionHandler newHttp2ConnectionHandler(ChannelPipeline pipelin
         return handler;
     }
 
+    private final class ProtocolDetectionHandler extends ByteToMessageDecoder {
+
+        private final EnumSet<SessionProtocol> candidates;
+        @Nullable
+        private final EnumSet<SessionProtocol> nextCandidates;
+        @Nullable
+        private final ProxiedAddresses proxiedAddresses;
+
+        ProtocolDetectionHandler(Set<SessionProtocol> protocols, @Nullable ProxiedAddresses proxiedAddresses) {
+            candidates = EnumSet.copyOf(protocols);
+            if (protocols.contains(PROXY)) {
+                // NB: We copy from 'candidates' rather than from 'protocols'
```

Comment:
This seems somewhat obvious can consider leaving out.

Reply:
Removed :-)